Neos.ContentRepository.Core/Classes/Feature/Common/ContentStreamIdOverride.php

```diff
@@ -24,13 +27,41 @@ class ContentStreamIdOverride
     /**
      * A content stream id that to use instead of the workspace one's {@see ConstraintChecks::requireContentStream()}
      */
-    public static ?ContentStreamId $contentStreamIdToUse = null;
+    private static ?ContentStreamId $contentStreamIdToUse = null;
 
     /**
      * @internal
      */
-    public static function useContentStreamId(?ContentStreamId $contentStreamIdToUse): void
+    public static function withContentStreamIdToUse(ContentStreamId $contentStreamIdToUse, \Closure $fn): void
     {
+        if (self::$contentStreamIdToUse !== null) {
+            throw new \Exception('Recursive content stream override is not supported', 1710426945);
+        }
         self::$contentStreamIdToUse = $contentStreamIdToUse;
+        try {
+            $fn();
+        } catch (\Throwable $th) {
+            self::$contentStreamIdToUse = null;
+            throw $th;
+        }
+        self::$contentStreamIdToUse = null;
+    }
+
+    /**
+     * @internal
+     */
+    public static function findContentStreamIdForWorkspace(ContentRepository $contentRepository, WorkspaceName $workspaceName): ContentStreamId
```

Comment:
nitpick: We have the guideline to use `find*` for methods that can return `null`
```suggestion
    public static function getContentStreamIdForWorkspace(ContentRepository $contentRepository, WorkspaceName $workspaceName): ContentStreamId
```